### PR TITLE
Implement browse schools

### PR DESCRIPTION
**Pull request by kevinjihwanlee**
screenshot 2018-11-20 20 29 28

### CODE REVIEW

**Comment by kevinjihwanlee** [comment]
[Implement browse schools](https://trello.com/c/tjNYggL4/50-implement-browse-schools)

File: src/components/App.js [hunk 1]
```diff
@@ -12,27 +15,45 @@ export const styles = {
...
+  fetchSchools: () => {
+    dispatch(fetchSchools('4720'));
+  },
+});
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.classes = props.classes;
+    this.fetchSchools = props.fetchSchools;
```
Comment (Joonpark13) [inline src/components/App.js]: You rarely ever want to set state using props. Here, there's no need to set any state - you don't even need a constructor. Just use the props themselves in the render & lifecycle method.

File: src/components/App.js [hunk 2]
```diff
@@ -12,27 +15,45 @@ export const styles = {
...
+  }
+
+  componentDidMount() {
+    this.fetchSchools();
```
Comment (Joonpark13) [inline src/components/App.js]: just call the prop function

File: src/components/App.js [hunk 3]
```diff
@@ -12,27 +15,45 @@ export const styles = {
...
+  }
 
-        <Grid item md={3} sm={12} xs={12}>
-          <SidebarView />
+  render() {
+    return (
+      <React.Fragment>
+        <TopBar />
+        <Grid container>
+          <Grid item md={9} sm={12} xs={12} className={this.classes.calendar}>
```
Comment (Joonpark13) [inline src/components/App.js]: just use the `classes` prop

File: src/components/App.js [hunk 4]
```diff
@@ -12,27 +15,45 @@ export const styles = {
...
 
 App.propTypes = {
   classes: PropTypes.objectOf(PropTypes.string).isRequired,
+  fetchSchools: PropTypes.func.isRequired,
 };
 
 export { App as UnstyledApp };
 
-export default withStyles(styles)(App);
+export default connect(null, mapDispatchToProps)(toJS(withStyles(styles)(App)));
```
Comment (Joonpark13) [inline src/components/App.js]: Just for clarity's sake, can you break this out into two lines?
`const AppWithStyles = withStyles(styles)(App);`
`export default connect(null, mapDispatchToProps)(AppWithStyles);`

File: src/components/App.test.js [hunk 5]
```diff
@@ -4,9 +4,11 @@ import { mockStyles } from 'util/testing';
 import { UnstyledApp, styles } from './App';
 
 describe('App', () => {
+  const dummyFunction = () => null;
+
   it('should render correctly', () => {
     const classes = mockStyles(styles);
-    const wrapper = shallow(<UnstyledApp classes={classes} />);
+    const wrapper = shallow(<UnstyledApp classes={classes} fetchSchools={dummyFunction} />);
```
Comment (Joonpark13) [inline src/components/App.test.js]: You can just give it an empty anonymous function: `() => {}`

File: src/components/App.js [hunk 6]
```diff
@@ -12,27 +15,45 @@ export const styles = {
...
 
-function App({ classes }) {
-  return (
-    <React.Fragment>
-      <TopBar />
-      <Grid container>
-        <Grid item md={9} sm={12} xs={12} className={classes.calendar}>
-          <CalendarViewContainer />
-        </Grid>
+export const mapDispatchToProps = dispatch => ({
```
Comment (Joonpark13) [inline src/components/App.js]: Put the mapDispatchToProps at the bottom of the file, just before the `export`s to follow convention.

File: src/components/App.js [hunk 7]
```diff
@@ -12,27 +15,45 @@ export const styles = {
...
+  fetchSchools: () => {
```
Comment (Joonpark13) [inline src/components/App.js]: You can use [shorthand notation](https://egghead.io/lessons/javascript-redux-using-mapdispatchtoprops-shorthand-notation) for this.

**Review by Joonpark13 — CHANGES_REQUESTED** [review]

File: src/components/App.js (hunk 1 above)
Comment (Joonpark13) [reply src/components/App.js]: This will also fix your coverage issue.

File: src/components/App.js (hunk 7 above)
Comment (Joonpark13) [reply src/components/App.js]:
```javascript
const mapDispatchToProps = {
  fetchSchools,
};
```

**Review by Joonpark13 — APPROVED** [review]

**Pull request merged**